### PR TITLE
TEC-6464 convert string to Linkedin URN

### PR DESCRIPTION
**Pull request by eddspencer**
### Description of Changes

Modified the default json mapper to handle serialization and deserialization of URNs.

### Documentation

N/A

### Risks & Impacts

low risk for now but consequent change to how the json mapper handles strings

### Testing

unit tested for both serialization and deserialization

### Compare (For layered PRs)
N/A

### CODE REVIEW

**Comment by eddspencer** [comment]
Nice one Clement :+1:

**Comment by eddspencer** [comment]
@diepjy I cant override these PRs and merge them, could you please do the honours?

File: src/main/java/com/echobox/api/linkedin/jsonmapper/DefaultJsonMapper.java [hunk 1]
```diff
@@ -263,6 +265,11 @@ public DefaultJsonMapper(JsonMappingErrorHandler jsonMappingErrorHandler) {
         return null;
       }
 
+      if (URN.class.equals(type)) {
```
Comment (diepjy) [inline src/main/java/com/echobox/api/linkedin/jsonmapper/DefaultJsonMapper.java]: I think this should be in `DefaultJsonMapper.toJavaType`?
Happy to discuss

File: src/main/java/com/echobox/api/linkedin/types/urn/URN.java [hunk 2]
```diff
@@ -20,16 +20,92 @@
...
+    this.id = id;
+  }
+
+  /**
+   * Instantiates a new Urn.
+   *
+   * @param urnString the urn string
+   */
+  public URN(String urnString) {
+    if (!urnString.startsWith("urn:li:")) {
```
Comment (diepjy) [inline src/main/java/com/echobox/api/linkedin/types/urn/URN.java]: Should have a null check to ensure `urnString` is not blank?

File: src/main/java/com/echobox/api/linkedin/jsonmapper/DefaultJsonMapper.java (hunk 1 above)
Comment (eddspencer) [reply src/main/java/com/echobox/api/linkedin/jsonmapper/DefaultJsonMapper.java]: good point, in fact taking a look I have managed to simplify the conversion further.

**Pull request merged**